### PR TITLE
feat: introduce DOMPolicy for stronger security throughout templating

### LINKED ISSUE #6447: feat: introduce DOMPolicy and plumb it throughout templating
We have identified a number of improvements we can make to fast-element, which will make it safer to use for developers, particularly those that are less aware of certain dangers related to HTML DOM sinks which can result in XSS attacks.

This set of improvements will introduce a new `DOMPolicy` abstraction which will be consumed by the template compilation and binding APIs. Through this abstraction, application developers can plug in custom trusted types and sanitizers across the entire templating and binding surface area. They will be able to define a default policy as well as per-binding policies. Additionally, the new API will make it easy too integrate libraries like [safevalues](https://github.com/google/safevalues).

### PR DESCRIPTION
**Pull request by EisenbergEffect**
# Pull Request

## 📖 Description

This PR introduces a new `DOMPolicy` interface which is piped through every template, directive, and binding. Policies are used to control HTML creation as well as access to all DOM sinks.

### 🎫 Issues

* Closes #6447 

## 👩‍💻 Reviewer Notes

Coming...

## 📑 Test Plan

All existing tests continue to pass. I have also added tests for directives, bindings, templates, etc. that exercise the DOM Policy capabilities. I've also added basic tests for the policy creation helper.

Perf benchmarks are a bit slower in a few cases. That is not unexpected when introducing something like this. However, the changes are minimal and I don't have real concerns about this.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

## ⏭ Next Steps

While working on this PR I noticed a few additional areas internally that could use some cleanup. I will follow up with an additional PR to take care of that.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
This PR is pretty much code complete now but needs additional tests before we merge. I'll open it up for review once the remaining tests are finished. It shouldn't be long as the core already has tests; I just need to add tests for the policy creation helper.

**Comment by EisenbergEffect** [comment]
This PR is now ready for review.

File: packages/web-components/fast-element/src/templating/binding-two-way.ts [hunk 1]
```diff
@@ -158,13 +159,15 @@ class TwoWayBinding<TSource = any, TReturn = any, TParent = any> extends Binding
  * Creates a default binding.
  * @param expression - The binding to refresh when changed.
  * @param optionsOrChangeEvent - The binding options or the name of the change event to use.
+ * @param policy - The security policy to associate with th binding.
```
Comment (nicholasrice) [inline packages/web-components/fast-element/src/templating/binding-two-way.ts:162]:
```suggestion
 * @param policy - The security policy to associate with the binding.
```

**Review by nicholasrice — APPROVED** [review]
Looks great, thanks @EisenbergEffect!

**Review by chrisdholt — APPROVED** [review]
Thanks @eisenbergeffect, LGTM!

**Pull request merged**